app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java

```diff
@@ -0,0 +1,181 @@
+        if (instance == null) {
+            instance = new SMSContactManager(applicationContext);
+        }
+        return instance;
+    }
+
+    //---------------------------- OPERATIONS ON THE SINGLE CONTACT ----------------------------
+
+    /**
+     * Verify if a contact could be a valid contact in its phone
+     *
+     * @param contactPhone  Contact phone to verify
+     *
+     * @return true if contact is valid, false otherwise
+     */
+    public boolean isValidContactPhone(@NonNull String contactPhone){
+        try {
+            new SMSPeer(contactPhone);
+        }catch (InvalidTelephoneNumberException addressException){
+            return false;
+        }
+        return true;
+    }
+
+    //---------------------------- OPERATIONS ON THE CONTACTS DATABASE ----------------------------
+
+    /**
+     * Add a {@link SMSPeer} as {@link SMSContact} in {@link SMSContactDatabase}
+     * after using {@link SMSContactConverterUtils} to convert SMSPeer in a Contact entity
+     *
+     * @param peer      {@link SMSPeer} to insert in the contacts database
+     */
+    public void addContact(@NonNull SMSPeer peer) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer);
+        contactDatabase.access().insert(newContact);
+    }
+
+    /**
+     * Add a {@link SMSPeer} as {@link SMSContact} in {@link SMSContactDatabase}
+     * after using {@link SMSContactConverterUtils} to convert SMSPeer in a Contact entity
+     *
+     * @param peer        {@link SMSPeer} to insert in the contacts database
+     * @param nameContact Optional name for the new contact
+     */
+    public void addContact(@NonNull SMSPeer peer, @NonNull String nameContact) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer, nameContact);
+        contactDatabase.access().insert(newContact);
+    }
+
+    /**
+     * Add a {@link SMSContact} in {@link SMSContactDatabase}
+     *
+     * @param newContact    {@link SMSContact} to insert in the contacts database
+     */
+    public void addContact(@NonNull SMSContact newContact) {
+        contactDatabase.access().insert(newContact);
+    }
+
+    /**
+     * Modify name of a contact
+     *
+     * @param peerToModify  {@link SMSPeer} represents the address of contact to modify
+     * @param newName       New name for the existing contact
+     */
+    public void modifyContactName(@NonNull SMSPeer peerToModify, @NonNull String newName){
+        SMSContact contact = SMSContactConverterUtils.contactFromPeer(peerToModify, newName);
+        contactDatabase.access().update(contact);
+    }
+
+    /**
+     * Remove a {@link SMSPeer} from {@link SMSContactDatabase}
+     *
+     * @param peer      {@link SMSPeer} to delete from the contacts database
+     */
+    public void removeContact(@NonNull SMSPeer peer) {
+        SMSContact oldContact = SMSContactConverterUtils.contactFromPeer(peer);
+        contactDatabase.access().delete(oldContact);
```

Comment:
Same thing here too, advantages are not evident now but otherwise it is a possible source of bugs in the future.
```suggestion
        removeContact(oldContact);
```